src/movies.js

```diff
@@ -1,25 +1,128 @@
 // Iteration 1: All directors? - Get the array of all directors.
+
+// function getAllDirectors(movies){
+//     const directorsListing = movies.map(function (moviesObj){
+//     return moviesObj.director;
+    
+// });
+//     return directorsListing
+// };
+
+function getAllDirectors(moviesArray) {
+    
+     const directorsArray = moviesArray.map(function(movieObj){
+    return movieObj.director;
+ });
+
+return directorsArray
+}
+
+
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+//function getAllDirectors(moviesArray) {}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+// function howManyMovies(moviesArray) {
+
+//     const dramaBySpielberg = moviesArray.filter(function(movieObJ {
+//         if (movieObj.director === "Spielberg" && movieObj.genre.includes("Drama")) {
+//             return true;
+//         } else {
+//             return false;
+//         }
+//         }));
+//         return dramaBySpielberg.length;
+//     };
+
+// function howManyMovies(moviesArray) {
+//   if (moviesArray.length) === 0) {
+//         return 0;
+//     }
+  
+//     const dramaBySpielberg = moviesArray.filter(function(movieObj) {
+
+//     if (movieObj.director === "Spielberg" && movieObj.genre.includes("Drama")) {
+//       return true;
+//     } else {
+//       return false;
+//     }
+//   });
+
+//   return dramaBySpielberg.length;
+// };
+
+function howManyMovies(moviesArray) {
```

Comment:
For that one, because you're using a count, probably it would be easier to use the `.forEach()` method.

Another approach would be to `return movieObj.director === 'Steven Spielberg' && movieObj.genre.includes('Drama')` inside the filter, so that way it would include in the final array all the movies that have the Steven Spielberg as Director and is a Drama movie.

and because we need a number as the return we can return the length of that filter for or function `return dramaBySpielberg.length`

Putting everything together:
```
function howManyMovies(moviesArray) {
	const dramaBySpielberg = moviesArray.filter(function (movieObj) {
		return movieObj.director === 'Steven Spielberg' && movieObj.genre.includes('Drama')
	})

	return dramaBySpielberg.length
}
```